src/helper.js

```diff
@@ -170,7 +170,32 @@ export const cloneWithProject = (sourceDm, projField, config, allFields) => {
     return cloned;
 };
 
+const sanitizeSchema = (schema) => {
+    // Do deep clone of the schema as the user might change the input schema reference.
+    const clonedSchema = JSON.parse(JSON.stringify(schema));
```

Comment:
`numberFormat` might be a function which won't be serialized during the above conversion.